cfme/configure/configuration/server_settings.py

```diff
@@ -451,15 +451,279 @@ def _save_action(self, view, updates, reset):
+    @property
+    def is_displayed(self):
+        return (
+            self.authentication_mode.is_displayed and
+            self.authentication_mode.selected_option == 'Database'
+        )
+
+
+class LdapAuthenticationView(ServerAuthenticationView):
+    """ Ldap Authentication View """
+
+    ldap_host_1 = Input(name='authentication_ldaphost_1')
+    ldap_host_2 = Input(name='authentication_ldaphost_2')
+    ldap_host_3 = Input(name='authentication_ldaphost_3')
+    port = Input(name='authentication_ldapport')
+    user_type = BootstrapSelect(id='authentication_user_type')
+    domain_prefix = Input(name='authentication_domain_prefix')
+    user_suffix = Input(name='authentication_user_suffix')
+
+    get_roles = Checkbox(name='ldap_role')
+    get_groups = Checkbox(name='get_direct_groups')
+    follow_referrals = Checkbox(name='follow_referrals')
+    base_dn = Input(name='authentication_basedn')
+    bind_dn = Input(name='authentication_bind_dn')
+    bind_password = Input(name='authentication_bind_pwd')
+    validate = Button('Validate')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.authentication_mode.is_displayed and
+            self.authentication_mode.selected_option == 'LDAP'
+        )
+
+
+class LdapsAuthenticationView(LdapAuthenticationView):
+    """ Ldaps Authentication View """
+
+    @property
+    def is_displayed(self):
+        return (
+            self.authentication_mode.is_displayed and
+            self.authentication_mode.selected_option == 'LDAPS'
+        )
+
+
+class AmazonAuthenticationView(ServerAuthenticationView):
+    """ Amazon Authentication View """
+
+    access_key = Input(name='authentication_amazon_key')
+    secret_key = Input(name='authentication_amazon_secret')
+
+    get_groups = Checkbox(name='amazon_role')
+    validate = Button('Validate')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.authentication_mode.is_displayed and
+            self.authentication_mode.selected_option == 'Amazon'
+        )
+
+
+class ExternalAuthenticationView(ServerAuthenticationView):
+    """ External Authentication View """
+
+    enable_sso = Checkbox(name='sso_enabled')
+    enable_saml = Checkbox(name='saml_enabled')
+
+    get_groups = Checkbox(name='httpd_role')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.authentication_mode.is_displayed and
+            self.authentication_mode.selected_option == 'External (httpd)'
+        )
+
+
+class AuthenticationSetting(NavigatableMixin, Updateable, Pretty):
+    """
+        Represents Authentication Setting for CFME
+
+        Args:
+            auth_mode: authorization mode, default value 'Database'
+    """
+
+    pretty_attrs = ['auth_mode']
+
+    user_type_dict = {
+        'userprincipalname': 'User Principal Name',
+        'dn-uid': 'Distinguished Name (UID=<user>)'
+    }
+
+    def __init__(self, appliance, auth_mode='Database'):
```

Comment:
**Required**
Don't use mutable string for arg default, use `None` and set auth_mode to 'Database' string within _init_ if its None.

Reply:
string isn't mutable... but I'd still rather see `None` used :)